src/test/java/htsjdk/samtools/HtsgetBAMFileReaderTest.java

```diff
@@ -0,0 +1,255 @@
+package htsjdk.samtools;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.util.CloseableIterator;
+import htsjdk.samtools.util.CoordMath;
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+
+public class HtsgetBAMFileReaderTest extends HtsjdkTest {
+    private static final String HTSGET_ENDPOINT = "http://127.0.0.1:3000/reads/";
+    private static final String LOCAL_PREFIX = "htsjdk_test.";
+
+    private final static URI htsgetBAM = URI.create(HTSGET_ENDPOINT + LOCAL_PREFIX + "index_test.bam");
+
+    private final static File bamFile = new File("src/test/resources/htsjdk/samtools/BAMFileIndexTest/index_test.bam");
+    private final static File csiFileIndex = new File(bamFile.getPath() + ".csi");
+
+    private final static int nofMappedReads = 9721;
+    private final static int nofUnmappedReads = 279;
+    private final static int noChrMReads = 23;
+    private final static int noChrMReadsContained = 9;
+    private final static int noChrMReadsOverlapped = 10;
+
+    private static HtsgetBAMFileReader bamFileReaderHtsget;
+    private static BAMFileReader bamFileReaderCSI;
+
+    @BeforeTest
+    public void init() throws IOException {
+        bamFileReaderHtsget = new HtsgetBAMFileReader(htsgetBAM, true, ValidationStringency.DEFAULT_STRINGENCY, DefaultSAMRecordFactory.getInstance(), false);
+        bamFileReaderCSI = new BAMFileReader(bamFile, csiFileIndex, true, false, ValidationStringency.DEFAULT_STRINGENCY, DefaultSAMRecordFactory.getInstance());
+    }
+
+    @Test
+    public static void testGetHeader() {
+        final SAMFileHeader expectedHeader = bamFileReaderCSI.getFileHeader();
+        final SAMFileHeader actualHeader = bamFileReaderHtsget.getFileHeader();
+        Assert.assertEquals(actualHeader, expectedHeader);
+    }
+
+    @Test
+    public static void testQueryMapped() throws IOException {
+        try (final SamReader samReader = SamReaderFactory.makeDefault().open(bamFile);
+             final SAMRecordIterator samRecordIterator = samReader.iterator()) {
+            Assert.assertEquals(samReader.getFileHeader().getSortOrder(), SAMFileHeader.SortOrder.coordinate);
+
+            int counter = 0;
+            while (samRecordIterator.hasNext()) {
+                final SAMRecord samRecord = samRecordIterator.next();
+                if (samRecord.getReferenceIndex() == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
+                    break;
+                }
+                if (counter++ % 100 > 1) { // test only 1st and 2nd in every 100 to speed the test up:
```

Comment:
How slow is it without this?

Reply:
It takes a few minutes before actually causing the process inside the docker container to hang so I think keeping this would be a good idea. The test including the skip was copied from the BAMFileReaderTest so presumably it was taking a while there too